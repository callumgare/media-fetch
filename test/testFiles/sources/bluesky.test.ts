import { expect } from "vitest";
import blueskySource from "@/src/plugins/built-in-sources/bluesky/index.js";
import { createBasicTestsForRequestHandlers } from "../../utils/vitest.js";

createBasicTestsForRequestHandlers({
  source: blueskySource,
  queries: {
    "single-media": {
      request: {
        id: "at://did:plc:z72i7hdynmk6r22z27h6tvur/app.bsky.feed.post/3jt6walwmos2y#bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi",
      },
      checkResponse: (response) => expect(response).toMatchSnapshot(),
    },
    search: {
      request: { searchText: "#photo" },
      checkResponse: (response) =>
        expect(response.media.length).toBeGreaterThan(2),
      numOfPagesToLoad: 1,
      duplicateMediaPossible: true,
    },
    feed: {
      request: {
        feedId:
          "at://did:plc:z72i7hdynmk6r22z27h6tvur/app.bsky.feed.generator/whats-hot",
      },
      checkResponse: (response) =>
        expect(response.media.length).toBeGreaterThan(2),
      numOfPagesToLoad: 2,
      duplicateMediaPossible: true,
      timeout: 10 * 1000,
    },
  },
});
