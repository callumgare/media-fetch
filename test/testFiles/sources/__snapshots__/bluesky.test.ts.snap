// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Run query "Single media" with: { "id": "at://did:plc:z72i7hdynmk6r22z27h6tvur/app.bsky.feed.post/3jt6walwmos2y#bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi" } 1`] = `
{
  "media": [
    {
      "contentHash": "bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi",
      "dateUploaded": 2023-04-12T17:36:42.427Z,
      "files": [
        {
          "aspectRatio": undefined,
          "ext": "jpeg",
          "fileSize": 148647,
          "image": true,
          "mimeType": "image/jpeg",
          "type": "full",
          "url": "https://cdn.bsky.app/img/feed_fullsize/plain/did:plc:z72i7hdynmk6r22z27h6tvur/bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi@jpeg",
          "video": false,
        },
        {
          "aspectRatio": undefined,
          "ext": "jpeg",
          "fileSize": undefined,
          "image": true,
          "mimeType": "image/jpeg",
          "type": "thumbnail",
          "url": "https://cdn.bsky.app/img/feed_thumbnail/plain/did:plc:z72i7hdynmk6r22z27h6tvur/bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi@jpeg",
          "video": false,
        },
      ],
      "id": "at://did:plc:z72i7hdynmk6r22z27h6tvur/app.bsky.feed.post/3jt6walwmos2y#bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi",
      "mediaFinderSource": "bluesky",
      "nameOfUploader": "Bluesky",
      "title": "How it started:",
      "url": "https://bsky.app/profile/did:plc:z72i7hdynmk6r22z27h6tvur/post/3jt6walwmos2y",
      "usernameOfUploader": "bsky.app",
    },
  ],
  "request": {
    "id": "at://did:plc:z72i7hdynmk6r22z27h6tvur/app.bsky.feed.post/3jt6walwmos2y#bafkreidf3ystxebv33boyb5hr2ggwmlee5x53vubwelsuvmcw3tq4p36pi",
    "queryType": "single-media",
    "source": "bluesky",
  },
}
`;
